src/dmd/root/array.d

```diff
@@ -591,3 +591,81 @@ unittest
     static assert (!is(typeof(Array!(OtherStruct).init.sort())));
     static assert (!is(typeof(Array!(OtherStruct).init.sort!pred)));
 }
+
+/**
+ * Iterates the given array and calls the given callable for each element.
+ *
+ * Use this instead of `foreach` when the array may expand during iteration.
```

Comment:
This should also refer to `foreachDsymbol` and vice-versa.

Also would appreciate a DDOC link.

Reply:
Sure.

Reply:
Done.